spring-webflux/src/main/java/org/springframework/web/reactive/socket/client/JettyWebSocketClient.java

```diff
@@ -0,0 +1,102 @@
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.web.reactive.socket.client;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.eclipse.jetty.client.Request;
+import org.eclipse.jetty.client.Response;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.util.component.LifeCycle;
+import org.eclipse.jetty.websocket.client.ClientUpgradeRequest;
+import org.eclipse.jetty.websocket.client.JettyUpgradeListener;
+import reactor.core.publisher.Mono;
+import reactor.core.publisher.Sinks;
+
+import org.springframework.core.io.buffer.DefaultDataBufferFactory;
+import org.springframework.http.HttpHeaders;
+import org.springframework.lang.Nullable;
+import org.springframework.web.reactive.socket.HandshakeInfo;
+import org.springframework.web.reactive.socket.WebSocketHandler;
+import org.springframework.web.reactive.socket.adapter.JettyWebSocketHandlerAdapter;
+import org.springframework.web.reactive.socket.adapter.JettyWebSocketSession;
+
+public class JettyWebSocketClient implements WebSocketClient {
+
+	private final org.eclipse.jetty.websocket.client.WebSocketClient client;
+
+	public JettyWebSocketClient() {
+		this.client = new org.eclipse.jetty.websocket.client.WebSocketClient();
+		LifeCycle.start(this.client);
+	}
+
+	public JettyWebSocketClient(org.eclipse.jetty.websocket.client.WebSocketClient client) {
+		this.client = client;
+	}
+
+	public void start() throws Exception {
+		this.client.start();
+	}
+
+	public void stop() throws Exception {
+		this.client.stop();
+	}
+
+	@Override
+	public Mono<Void> execute(URI url, WebSocketHandler handler) {
+		return execute(url, null, handler);
+	}
+
+	@Override
+	public Mono<Void> execute(URI url, @Nullable HttpHeaders headers, WebSocketHandler handler) {
+
+		ClientUpgradeRequest upgradeRequest = new ClientUpgradeRequest();
+		upgradeRequest.setSubProtocols(handler.getSubProtocols());
+		if (headers != null) {
+			headers.keySet().forEach(header -> upgradeRequest.setHeader(header, headers.getValuesAsList(header)));
+		}
+
+		AtomicReference<HandshakeInfo> handshakeInfo = new AtomicReference<>();
+		JettyUpgradeListener jettyUpgradeListener = new JettyUpgradeListener() {
+			@Override
+			public void onHandshakeResponse(Request request, Response response) {
+				String protocol = response.getHeaders().get(HttpHeader.SEC_WEBSOCKET_SUBPROTOCOL);
+				HttpHeaders responseHeaders = new HttpHeaders();
+				response.getHeaders().forEach(header -> responseHeaders.addAll(header.getName(), header.getValueList()));
```

Comment:
This is broken for `Set-Cookie` with the `Expires` attribute.
See https://github.com/cometd/cometd/commit/9217e1b124211732943ff18456da677a354fd14f, unfortunately.

Reply:
Is that a bug in Jetty?

I will change this to `responseHeaders.add(header.getName(), header.getValue())`.